CONTRIBUTING.md

```diff
@@ -0,0 +1,280 @@
+/// @arg {real} oldvalue    The value to re-interpolate.
+/// @arg {real} newfrom     The new range's first value.
+/// @arg {real} newto       The new range's second value.
+/// @returns {real}
+/// @desc Given an interpolated value in the old range (oldfrom, oldto), returns its equivalent value in the new range (newfrom, newto).
+```
+
+## Naming Conventions
+
+### Function Names
+
+- **Use `snake_case` for all function names.**
+- **Functions that perform some action should be phrased like commands.** Examples:
+	- `do_the_thing(...)`
+	- `calculate_some_result(...)`
+- **Functions that perform some checks should be phrased like questions.** Examples:
+	- `is_defined(...)`
+	- `has_items(...)`
+	- `can_move(...)`
+- Functions that share a domain with built-in functions should adopt the same prefix as them whenever possible so that they are easier to find. Examples:
+	- `file_read_all_text(...)`
+	- `draw_arc(...)`
+- Functions that expand upon some built-in functionality of GameMaker should use naming similar to the original domain. Example: an array-based counterpart to `collision_circle_list` would be called `collision_circle_array`.
+- When making a toolbox function similar to some industry standard, it's worth checking the commonly accepted naming scheme. Since GML borrows lots of naming conventions from JavaScript, JavaScript conventions will take precedence over other languages in case of multiple competing conventions.
+- Functions using these keywords in their names should have these expected behaviours:
+    - `get`: The function will return a result without modifying anything (except maybe local variables to obtain some result).
+    - `set`: When "set" is understood as an operation rather than a collection of items, the function is expected to change some variable and is unlikely to return a result.
+    - `load`: The function will read something from a file.
+    - `save`: The function will write/overwrite some file with a given content.
+    - `parse`: The function will read some value from a string.
+
+### Function Parameters and Local Variables
+
+- **Prepend an underscore (`_`) to the names of local variables**, in order to avoid collisions with keywords and instance/struct variables.
+
+**YES:**
+
+```gml
+function relerp(_oldfrom, _oldto, _oldvalue, _newfrom, _newto) {
+    if (_oldto == _oldfrom) {
+        throw "relerp() old range has zero width.";
+    }
+    var _amount = (_oldvalue - _oldfrom) / (_oldto - _oldfrom);
+    return _newfrom + (_newto - _newfrom) * _amount;
+}
+```
+
+**NO:**
+
+```gml
+function relerp(oldfrom, oldto, oldvalue, newfrom, newto) {
+    if (oldto == oldfrom) {
+        throw "relerp() old range has zero width.";
+    }
+    var amount = (oldvalue - oldfrom) / (oldto - oldfrom);
+    return newfrom + (newto - newfrom) * amount;
+}
+```
+
+### Toolbox Script Assets
+
+- **Use `utils_` as the prefix, and PascalCase for the rest of the name.** Example: `utils_CommunityToolboxScriptCategory`
```

Comment:
Recently, I considered expanding the prefix to `utils_CommunityToolbox` instead (so it would be `utils_CommunityToolboxString`, `utils_CommunityToolboxArray` etc.).

This would avoid collisions with project-specific utilities, and there's basically no reason why someone would refer to the script directly, so using a name `utils_CommunityToolboxString` rather than just `utils_String` shouldn't be a hassle.

Not sure if change to `utils_CommunityToolbox` scripts should be included alongside the contribution guidelines, or as a separate PR. Either way is fine with me.

Reply:
I'll update both the scripts and the Contribution Guidelines document at the same time, given your approval.